File: comparison/results/haiku/ponytail/humaneval/HumanEval_129/python/solution.py
def minPath(grid, k):
    n = len(grid)
    
    # Find position of value 1 (optimal starting point)
    start_pos = None
    for i in range(n):
        for j in range(n):
            if grid[i][j] == 1:
                start_pos = (i, j)
                break
        if start_pos:
            break
    
    best_path = None
    
    def dfs(i, j, path):
        nonlocal best_path
        
        if len(path) == k:
            if best_path is None or path < best_path:
                best_path = path[:]
            return
        
        # Pruning: if current path is worse than best path, stop
        if best_path is not None and path > best_path[:len(path)]:
            return
        
        # Try all 4 neighbors
        for di, dj in [(0, 1), (0, -1), (1, 0), (-1, 0)]:
            ni, nj = i + di, j + dj
            if 0 <= ni < n and 0 <= nj < n:
                path.append(grid[ni][nj])
                dfs(ni, nj, path)
                path.pop()
    
    dfs(start_pos[0], start_pos[1], [1])
    
    return best_path

File: comparison/results/haiku/ponytail/humaneval/HumanEval_129/typescript/solution.ts
function minPath(grid: number[][], k: number): number[] {
    const n = grid.length;
    
    // Find the position of value 1
    let startRow = 0, startCol = 0;
    for (let i = 0; i < n; i++) {
        for (let j = 0; j < n; j++) {
            if (grid[i][j] === 1) {
                startRow = i;
                startCol = j;
            }
        }
    }
    
    let result: number[] | null = null;
    
    function dfs(row: number, col: number, path: number[]): void {
        // Pruning: if current path is already worse than result
        if (result !== null) {
            for (let i = 0; i < path.length; i++) {
                if (path[i] > result[i]) {
                    return;  // Prune: current path is lexicographically larger
                }
                if (path[i] < result[i]) {
                    break;  // Current path is potentially better, continue exploring
                }
            }
        }
        
        // If we've reached length k, update result
        if (path.length === k) {
            if (result === null) {
                result = [...path];
            }
            return;
        }
        
        // Explore neighbors
        const directions = [[0, 1], [1, 0], [0, -1], [-1, 0]];
        const neighbors: [number, number, number][] = [];
        
        for (const [dr, dc] of directions) {
            const newRow = row + dr;
            const newCol = col + dc;
            
            if (newRow >= 0 && newRow < n && newCol >= 0 && newCol < n) {
                neighbors.push([grid[newRow][newCol], newRow, newCol]);
            }
        }
        
        // Sort neighbors by value to explore smaller values first (optimization)
        neighbors.sort((a, b) => a[0] - b[0]);
        
        for (const [val, newRow, newCol] of neighbors) {
            path.push(val);
            dfs(newRow, newCol, path);
            path.pop();
        }
    }
    
    dfs(startRow, startCol, [1]);
    return result!;
}
